overlord/snapstate/snapmgr.go

```diff
@@ -73,10 +73,9 @@ func (ssfl *SnapSetupFlags) UnmarshalJSON(b []byte) error {
 
 // SnapSetup holds the necessary snap details to perform most snap manager tasks.
 type SnapSetup struct {
-	Name     string        `json:"name"`
-	Revision snap.Revision `json:"revision,omitempty"`
-	Channel  string        `json:"channel,omitempty"`
-	UserID   int           `json:"user-id,omitempty"`
+	// FIXME: rename to RequestedChannel to convey the meaning better
```

Comment:
The idea behing this was that we have "channel" in both SnapSetup and in SnapSetup.SideInfo.Channel  and that caused a bit of confusion in the review and I was wondering if we could find a way to make the distinction more obvious. Happy to leave as is if you think its already clear enough.